Directory.Build.targets

```diff
@@ -7,6 +7,21 @@
     <ImportDirectoryBuildTargets>false</ImportDirectoryBuildTargets>
   </PropertyGroup>
 
+  <!--
+  When .NET gets built from source, make the SDK aware there are bootstrap packages
+  for Microsoft.NETCore.App.Runtime.<rid> and Microsoft.NETCore.App.Crossgen2.<rid>.
+  -->
+  <ItemGroup Condition=" '$(DotNetBuildFromSource)' == 'true' ">
+    <KnownFrameworkReference Update="Microsoft.NETCore.App">
+      <RuntimePackRuntimeIdentifiers>%(KnownFrameworkReference.RuntimePackRuntimeIdentifiers);$(NETCoreSdkRuntimeIdentifier)</RuntimePackRuntimeIdentifiers>
+    </KnownFrameworkReference>
+    <KnownCrossgen2Pack Update="Microsoft.NETCore.App.Crossgen2">
+      <Crossgen2RuntimeIdentifiers>%(KnownCrossgen2Pack.Crossgen2RuntimeIdentifiers);$(NETCoreSdkRuntimeIdentifier)</Crossgen2RuntimeIdentifiers>
+    </KnownCrossgen2Pack>
+    <!-- Avoid references to Microsoft.AspNetCore.App.Runtime.<rid> -->
+    <KnownFrameworkReference Remove="Microsoft.AspNetCore.App" />
```

Comment:
This is a big hammer (even if it's only conditioned on source build). Can we instead condition this to only be removed when it would be used (i.e. when RuntimeIdentifier/s is used)?

Reply:
Also, any idea why this didn't show up earlier in code paths that use RuntimeIdentifier? I would have assumed that we already set this globally for source build, otherwise we would have that prebuilt showing up? cc @MichaelSimons

Reply:
We do a self-contained publish (using standard SDK support) for several tools in the repo now. The self-contained publish automatically references and restores ASP.NET runtime pack, even if you are just publishing a console app and do not need it.